lib/led/rgb.js

```diff
@@ -0,0 +1,189 @@
+module.exports = function(Led, priv) {
+
+  /**
+   * LedRGB
+   *
+   *
+   * @param  {[type]} opts [description]
+   * @return {[type]}      [description]
+   * @alias Led.RGB
+   */
+  var LedRGB = function(opts) {
+    if (!(this instanceof LedRGB)) {
+      return new LedRGB(opts);
+    }
+
+    var color, colors, k, state;
+
+    colors = LedRGB.colors.slice();
+    k = -1;
+
+    // This will normalize an array of pins in [ r, g, b ]
+    // order to an object that's shaped like:
+    // {
+    //   red: r,
+    //   green: g,
+    //   blue: b
+    // }
+    if (Array.isArray(opts.pins)) {
+      opts.pins = colors.reduce(function(pins, pin, i, list) {
+        return (pins[list[i]] = opts.pins[i], pins);
+      }, {});
+    }
+
+    // Initialize each Led instance
+    while (colors.length) {
+      color = colors.shift();
+      this[color] = new Led({
+        pin: opts.pins[color],
+        board: opts.board,
+        address: opts.address,
+        controller: opts.controller,
+        isAnode: opts.isAnode
+      });
+    }
+
+    this.interval = null;
+
+    state = {
+      red: 255,
+      green: 255,
+      blue: 255,
+      isAnode: opts.isAnode || false,
+      isRunning: false
+    };
+
+    priv.set(this, state);
+
+    Object.defineProperties(this, {
+      isOn: {
+        get: function() {
+          return LedRGB.colors.some(function(color) {
+            return this[color].isOn;
+          }, this);
+        }
+      },
+      isRunning: {
+        get: function() {
+          return state.isRunning;
+        }
+      },
+      isAnode: {
+        get: function () {
+          return state.isAnode;
+        }
+      },
+      values: {
+        get: function() {
+          return LedRGB.colors.reduce(function(current, color) {
+            return (current[color] = this[color].value, current);
+          }.bind(this), {});
+        }
+      }
+    });
+  };
+
+  LedRGB.colors = ["red", "green", "blue"];
+
+  /**
+   * color
+   *
+   * @param  {String} color Hexadecimal color string
+   * @param  {Array} color Array of color values
+   *
+   * @return {LedRGB}
+   */
+  LedRGB.prototype.color = function(value) {
```

Comment:
@boneskull @rwaldron Note that I just did a PR with new signature additions to Led.RGB.color() that will need to be copied here when/if that PR is merged. See #682